pylint/checkers/variables.py

```diff
@@ -1015,6 +1016,12 @@ def visit_name(self, node):
                 self._loopvar_name(node, name)
                 break
 
+            # the name has already been consumed, skip decorators
+            if name in current_consumer.consumed and utils.is_func_decorator(
+                current_consumer.node
```

Comment:
I think it should be an ``or`` line 1012 instead :)

Reply:
Thanks, moved